### ISSUE TITLE
total_batch_size//config.sample.num_batches_per_epoch == 0

### ISSUE THREAD

**Issue by Yovecent** [yifan123/flow_grpo#102]
When I was running the training code, I meet the problem below:

`total_batch_size//config.sample.num_batches_per_epoch == 0`   <br>

which will make the blow code to be wrong:

[this](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_sd3.py#L839)
```
# rebatch for training
samples_batched = {
    k: v.reshape(-1, total_batch_size//config.sample.num_batches_per_epoch, *v.shape[1:])
    for k, v in samples.items()

```

 I think the reason for this is the `mask.sum()  <  config.sample.num_batches_per_epoch`

So I think the code may add some condition to obmit it:

[start](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_sd3.py#L802)
```

# Get the mask for samples where all advantages are zero across the time dimension
mask = (samples["advantages"].abs().sum(dim=1) != 0)  # (16) (total_batch_size)

# If the number of True values in mask is not divisible by config.sample.num_batches_per_epoch,
# randomly change some False values to True to make it divisible
num_batches = config.sample.num_batches_per_epoch
true_count = mask.sum()

####⬇ HERE ⬇####
if true_count == 0:
    mask[0] = True
    true_count = 1
####⬆ HERE ⬆####

if true_count % num_batches != 0:
    false_indices = torch.where(~mask)[0]
    num_to_change = num_batches - (true_count % num_batches)
    if len(false_indices) >= num_to_change:
        random_indices = torch.randperm(len(false_indices))[:num_to_change]
        mask[false_indices[random_indices]] = True
if accelerator.is_main_process:
    wandb.log(
        {
            "actual_batch_size": mask.sum().item()//config.sample.num_batches_per_epoch,
        },
        step=global_step,
    )
# Filter out samples where the entire time dimension of advantages is zero
samples = {k: v[mask] for k, v in samples.items()}

total_batch_size, num_timesteps = samples["timesteps"].shape
# assert (
#     total_batch_size
#     == config.sample.train_batch_size * config.sample.num_batches_per_epoch
# )
assert num_timesteps == config.sample.num_steps


```
